src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -182,48 +187,78 @@ private boolean isTerminated(final DetailAST ast, boolean useBreak,
+                terminated =
+                        useBreak || hasLabel(ast, labelsForCurrentSwitchScope);
                 break;
             case TokenTypes.LITERAL_CONTINUE:
-                terminated = useContinue;
+                terminated =
+                        useContinue || hasLabel(ast, labelsForCurrentSwitchScope);
                 break;
             case TokenTypes.SLIST:
-                terminated = checkSlist(ast, useBreak, useContinue);
+                terminated =
+                        checkSlist(ast, useBreak, useContinue, labelsForCurrentSwitchScope);
                 break;
             case TokenTypes.LITERAL_IF:
-                terminated = checkIf(ast, useBreak, useContinue);
+                terminated =
+                        checkIf(ast, useBreak, useContinue, labelsForCurrentSwitchScope);
                 break;
             case TokenTypes.LITERAL_FOR:
             case TokenTypes.LITERAL_WHILE:
             case TokenTypes.LITERAL_DO:
-                terminated = checkLoop(ast);
+                terminated = checkLoop(ast, labelsForCurrentSwitchScope);
                 break;
             case TokenTypes.LITERAL_TRY:
-                terminated = checkTry(ast, useBreak, useContinue);
+                terminated =
+                        checkTry(ast, useBreak, useContinue, labelsForCurrentSwitchScope);
                 break;
             case TokenTypes.LITERAL_SWITCH:
-                terminated = checkSwitch(ast, useContinue);
+                terminated =
+                        checkSwitch(ast, useContinue, labelsForCurrentSwitchScope);
                 break;
             case TokenTypes.LITERAL_SYNCHRONIZED:
-                terminated = checkSynchronized(ast, useBreak, useContinue);
+                terminated =
+                        checkSynchronized(ast, useBreak, useContinue, labelsForCurrentSwitchScope);
+                break;
+            case TokenTypes.LABELED_STAT:
+                labelsForCurrentSwitchScope.add(ast.getFirstChild().getText());
+                terminated =
+                        isTerminated(ast.getLastChild(), useBreak, useContinue,
+                                labelsForCurrentSwitchScope);
                 break;
             default:
                 terminated = false;
         }
         return terminated;
     }
 
+    /**
+     * Checks if given break or continue ast has outer label.
+     *
+     * @param statement break or continue node
+     * @param labelsForCurrentSwitchScope the Set labels for the current scope of the switch
+     * @return true if local label used
+     */
+    private static boolean hasLabel(DetailAST statement, Set<String> labelsForCurrentSwitchScope) {
+        return Optional.ofNullable(statement)
+                .filter(node -> node.getChildCount() == 2)
+                .map(DetailAST::getFirstChild)
+                .map(DetailAST::getText)
+                .map(text -> !labelsForCurrentSwitchScope.contains(text))
```

Comment:
```suggestion
                .map(label -> !labelsForCurrentSwitchScope.contains(label))
```

Reply:
@relentless-pursuit was there an issue with filtering by TokenTypes.IDENT as suggested in https://github.com/checkstyle/checkstyle/pull/14199#discussion_r1525031818?

Reply:
@nrmancuso I missed it somehow. However, i see failed cases, and more violations expected than it is now. 

```
    private static boolean hasLabel(DetailAST statement, Set<String> labelsForCurrentSwitchScope) {
        return Optional.ofNullable(statement)
                .filter(child -> child.getType() == TokenTypes.IDENT)
                .map(DetailAST::getFirstChild)
                .map(DetailAST::getText)
                .map(text -> !labelsForCurrentSwitchScope.contains(text))
                .orElse(Boolean.FALSE);
    }
```

```
Violation lines for /Users/piyushkumarsadangi/Development/checkstyle/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/InputFallThroughLabeledBreak.java differ.
unexpected (7): 22, 33, 38, 64, 74, 83, 98
---
expected      : [51, 55, 69, 78, 88]
but was       : [22, 33, 38, 51, 55, 64, 69, 74, 78, 83, 88, 98]
```

```
    int method(int a, int b) {
        top: switch (a) {
            case 1:
                switch (b) {
                    case 10:
                        break top; // ok as terminates outer 'top' switch
                    default:
                        return 11;
                }
            case 2: // Expecting a Fall-through violation here, which shouldn't be the case
                return 2;
        }
        
        ...
```

Reply:
@nrmancuso , please ignore my above comment. I hurriedly tested without carefully observing your comment. 
Yes, it seems to be working. 

```
    private static boolean hasLabels(DetailAST statement, Set<String> labels) {
        return Optional.of(statement)
                .map(DetailAST::getFirstChild)  // get first child
                .filter(child -> child.getType() == TokenTypes.IDENT) // see if it is an identifier
                .map(DetailAST::getText) // if so, get the name
                .filter(label -> !labels.contains(label)) // see if this label is in labels
                .isPresent();
    }
```

I will push my code changes again with your suggestions. Thanks for your patience.